### ISSUE TITLE
Temperature Sensor Tube Module

### ISSUE THREAD

**Issue by johalun** [TeamPneumatic/pnc-repressurized#1258]
### Describe the feature

A tube module that reads the temperature of the block its facing and output a redstone signal to one of the 16 colored channels. It would have a simple function where it just outputs a signal strength based on hard coded min/max temperatures, and when upgraded with the advanced module, the redstone output can be configured just like the advanced regulator module.

### Reasons why it should be considered

Currently the only ways to measure temperature is with a programmable drone or the universal sensor. Both also capable of measuring temperature remotely without being near the block and are very different compared to tube modules. Having a temperature sensor that attaches to a tube facing the block to measure feels more inline with the existing tube modules and the existence and behavior of the temperature sensor tube module would feel like a natural part of the tube module ecosystem.

**Comment by johalun** [comment]
I might take a stab at coding a PoC this week. I can probably re-use bits from the RedstoneModule, RegulatorModule and PressureGaugeModule.

**Comment by FrederikRasmussen** [comment]
Just adding my support for this. Controlling temperature exclusively through regulator use can be near-impossible in some situations. A simple thermostat module would be wonderful.

**Comment by desht** [comment]
Yeah I'd be happy to add this.  @johalun feel free to have a go, let me know if you need any advice etc.

**Comment by johalun** [comment]
WIP at https://github.com/TeamPneumatic/pnc-repressurized/pull/1264 

I would appreciate some feedback on the work so far.
